source/core/module_coreHelpers.js

```diff
@@ -75,33 +104,68 @@
                     str += String.fromCharCode(u0);
                     continue;
                 }
+
+                // Look ahead to validate the UTF-8 structure
+                if ((u0 & 0xE0) === 0xC0) {
```

Comment:
However, I think a link to the spec or resource you followed to implement this, or inline comments with example encodings, would be helpful.